dbt_subprojects/hourly_spellbook/models/_project/balancer/erc4626_tokens/ethereum/balancer_v3_ethereum_erc4626_token_prices.sql

```diff
@@ -0,0 +1,62 @@
+{{ config(
+        schema = 'balancer_v3_ethereum',
+        alias = 'erc4626_token_prices',
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['minute', 'wrapped_token'],
+        incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.minute')]
+    )
+}}
+
+WITH wrap_unwrap AS(
+        SELECT 
+            evt_block_time,
+            wrappedToken,
+            CAST(depositedUnderlying AS DOUBLE) / CAST(mintedShares AS DOUBLE) AS ratio
+        FROM {{ source('balancer_v3_ethereum', 'Vault_evt_Wrap') }}
+        {% if is_incremental() %}
+        WHERE {{ incremental_predicate('evt_block_time') }}
+        {% endif %}      
+
+        UNION ALL
+
+        SELECT 
+            evt_block_time,
+            wrappedToken, 
+            CAST(withdrawnUnderlying AS DOUBLE) / CAST(burnedShares AS DOUBLE) AS ratio
+        FROM {{ source('balancer_v3_ethereum', 'Vault_evt_Unwrap') }}    
+        {% if is_incremental() %}
+        WHERE {{ incremental_predicate('evt_block_time') }}
+        {% endif %}      
+    ),
+
+
+    price_join AS(
+    SELECT 
+        w.evt_block_time,
+        m.underlying_token,
+        w.wrappedToken,
+        m.erc4626_token_symbol,
+        m.underlying_token_symbol,
+        m.decimals,
+        ratio * price AS adjusted_price
+    FROM wrap_unwrap w
+    JOIN {{ref('balancer_v3_ethereum_erc4626_token_mapping')}} m ON m.erc4626_token = w.wrappedToken
+    JOIN {{ source('prices', 'usd') }} p ON m.underlying_token = p.contract_address
+    AND p.blockchain = 'ethereum'
+    AND DATE_TRUNC('minute', w.evt_block_time) = DATE_TRUNC('minute', p.minute)
```

Comment:
looks like since you filter the `wrap_unwrap` data incrementally, should be able to apply incremental filter on prices join too?

Reply:
applies to gnosis version too

Reply:
great catch! just added